lib/aliquot-pay.rb

```diff
@@ -72,36 +72,63 @@ def self.four_byte_length(str)
     [str.length].pack('V')
   end
 
+  def self.generate_signature(*args)
+    args.map do |s|
+      four_byte_length(s) + s
+    end.join('')
+  end
+
   def self.signature_string(
     message,
-    recipient_id     = DEFAULTS[:merchant_id],
-    sender_id        = DEFAULTS[:info],
-    protocol_version = 'ECv1'
+    recipient_id:     "merchant:#{DEFAULTS[:merchant_id]}",
```

Comment:
Shouldn't this be a parameter (that defaults to `DEFAULTS[:merchant_id]`)?

(Likewise for `DEFAULTS[:info]` below.)

Reply:
Strike that. It's exactly what it is 🤦‍♂️